core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -1291,6 +1319,14 @@ object Stream {
   def constant[O](o: O, segmentSize: Int = 256): Stream[Pure, O] =
     segment(Segment.constant(o).take(segmentSize).voidResult).repeat
 
+  /**
+    * Returns a stream that when run, sleeps for duration `d` and then pulls from `s`.
+    *
+    * Alias for `sleep_[F](d) ++ s`.
+    */
+  def delay[F[_], O](s: Stream[F, O], d: FiniteDuration)(implicit F: Timer[F]): Stream[F, O] =
```

Comment:
I didn't really like this name in `Scheduler`, but in `Stream` is worse. I mean, it's perfectly clear what it does, and I can't think of an alternative, but it overlaps with `Sync.delay`....

Reply:
Yeah agreed -- I think we could remove this method entirely.

Reply:
Actually yeah, it could just be removed. The alternative with `sleep` is still super concise.

Reply:
An alternative name could be `delayFor`.

Reply:
Oh and this should move to a `Stream` instance method too. `s.sleepFor(5.seconds)`?

Reply:
Right, I think it could be a method on the Stream class, with that name.
`Stream(....).delayFor(3.seconds)`, or `delayBy`

Reply:
:+1: for `delayFor` or `delayBy` as instance method on `Stream`